hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FactoryIdHelper.java

```diff
@@ -18,154 +18,94 @@
-    public static final int MANAGEMENT_DS_FACTORY_ID = -24;
-
-    public static final String TEXT_PROTOCOLS_DS_FACTORY = "hazelcast.serialization.ds.text.protocols";
-    public static final int TEXT_PROTOCOLS_DS_FACTORY_ID = -25;
-
-    public static final String ENTERPRISE_HOTRESTART_BACKUP_DS_FACTORY = "hazelcast.serialization.ds.spi.hotrestart.backup";
-    public static final int ENTERPRISE_HOTRESTART_BACKUP_DS_FACTORY_ID = -26;
-
-    public static final String SCHEDULED_EXECUTOR_DS_FACTORY = "hazelcast.serialization.ds.scheduled.executor";
-    public static final int SCHEDULED_EXECUTOR_DS_FACTORY_ID = -27;
-
-    public static final String USER_CODE_DEPLOYMENT_DS_FACTORY = "hazelcast.serialization.ds.user.code.deployment";
-    public static final int USER_CODE_DEPLOYMENT_DS_FACTORY_ID = -28;
-
-    public static final String AGGREGATOR_DS_FACTORY = "hazelcast.serialization.ds.aggregator";
-    public static final int AGGREGATOR_DS_FACTORY_ID = -29;
-
-    public static final String PROJECTION_DS_FACTORY = "hazelcast.serialization.ds.projection";
-    public static final int PROJECTION_DS_FACTORY_ID = -30;
-
-    public static final String CONFIG_DS_FACTORY = "hazelcast.serialization.ds.config";
-    public static final int CONFIG_DS_FACTORY_ID = -31;
-
-    public static final String ENTERPRISE_SECURITY_DS_FACTORY = "hazelcast.serialization.ds.security";
-    public static final int ENTERPRISE_SECURITY_DS_FACTORY_ID = -32;
-
-    public static final String EVENT_JOURNAL_DS_FACTORY = "hazelcast.serialization.ds.event_journal";
-    public static final int EVENT_JOURNAL_DS_FACTORY_ID = -33;
-
-    public static final String FLAKE_ID_GENERATOR_DS_FACTORY = "hazelcast.serialization.ds.flake_id_generator";
-    public static final int FLAKE_ID_GENERATOR_DS_FACTORY_ID = -34;
-
-    public static final String SPLIT_BRAIN_DS_FACTORY = "hazelcast.serialization.ds.split_brain";
-    public static final int SPLIT_BRAIN_DS_FACTORY_ID = -35;
-
-    public static final String PN_COUNTER_DS_FACTORY = "hazelcast.serialization.ds.pn_counter";
-    public static final int PN_COUNTER_DS_FACTORY_ID = -36;
-
-    public static final String METRICS_DS_FACTORY = "hazelcast.serialization.metrics";
-    public static final int METRICS_DS_FACTORY_ID = -37;
-
-    public static final String SQL_DS_FACTORY = "hazelcast.serialization.sql";
-    public static final int SQL_DS_FACTORY_ID = -38;
-
-    public static final String JSON_DS_FACTORY = "hazelcast.serialization.json";
-    public static final int JSON_DS_FACTORY_ID = -39;
-
-    public static final String UTIL_COLLECTION_DS_FACTORY = "hazelcast.serialization.util.collection";
-    public static final int UTIL_COLLECTION_DS_FACTORY_ID = -40;
-
-    public static final String JET_SQL_DS_FACTORY = "hazelcast.serialization.jet.sql";
-    public static final int JET_SQL_DS_FACTORY_ID = -41;
+public final class FactoryIdHelper {
+    public enum Factory {
```

Comment:
I'm against this part of the design because you have no simple constant to refer to. 

Methods like getFactoryId should be able to do this and only this:

'return <STATIC_FINAL_CONST>'

Anything that is more complicated than that, is a no-go.